### PR TITLE
[ErrorRenderer] Improving the exception page provided by HtmlErrorRenderer

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

As part of the component responsability, this proposal is intended to provide a full exception page (advanced) in debug mode, just like the one provided by TwigBundle which would be deprecated.

### CODE REVIEW

**Comment by yceruto** [comment]
fabbot faults can be ignored? semicolon seems unnecessary.

File: src/Symfony/Component/ErrorRenderer/Resources/views/trace.html.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        </span>
+    <?php } ?>
+</div>
+<?php if ($trace['file']) { ?>
+    <div id="trace-html-<?= $prefix.'-'.$i ?>" class="trace-code sf-toggle-content">
+        <?= strtr($this->fileExcerpt($trace['file'], $trace['line'], 5), [
+            '#DD0000' => 'var(--highlight-string)',
+            '#007700' => 'var(--highlight-keyword)',
+            '#0000BB' => 'var(--highlight-default)',
+            '#FF8000' => 'var(--highlight-comment)'
```
Comment (Tobion) [inline src/Symfony/Component/ErrorRenderer/Resources/views/trace.html.php]: missing trailing comma

Comment (yceruto) [reply src/Symfony/Component/ErrorRenderer/Resources/views/trace.html.php]: fixed, thanks!

File: src/Symfony/Component/ErrorRenderer/ErrorRenderer/HtmlErrorRenderer.php [hunk 2]
```diff
@@ -53,276 +62,245 @@ public static function getFormat(): string
...
     /**
-     * Sets the format for links to source files.
-     *
-     * @param string|FileLinkFormatter $fileLinkFormat The format for links to source files
-     *
-     * @return string The previous file link format
+     * Gets the HTML content associated with the given exception.
      */
-    public function setFileLinkFormat($fileLinkFormat)
+    public function getBody(FlattenException $exception): string
```
Comment (Tobion) [inline src/Symfony/Component/ErrorRenderer/ErrorRenderer/HtmlErrorRenderer.php]: should be marked as internal

File: src/Symfony/Component/ErrorRenderer/ErrorRenderer/HtmlErrorRenderer.php [hunk 3]
```diff
@@ -53,276 +62,245 @@ public static function getFormat(): string
...
     }
 
     /**
-     * Gets the HTML content associated with the given exception.
-     *
-     * @return string The content as a string
+     * Gets the stylesheet associated with the given exception.
      */
-    public function getBody(FlattenException $exception)
+    public function getStylesheet(): string
```
Comment (Tobion) [inline src/Symfony/Component/ErrorRenderer/ErrorRenderer/HtmlErrorRenderer.php]: should be marked as internal

**Review by Tobion — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @yceruto.

**Pull request merged**